pkg/frontend/adminactions/kubeactions.go

```diff
@@ -130,3 +142,20 @@ func (k *kubeActions) KubeDelete(ctx context.Context, groupKind, namespace, name
 
 	return k.dyn.Resource(*gvr).Namespace(namespace).Delete(ctx, name, metav1.DeleteOptions{})
 }
+
+func (k *kubeActions) KubeDrain(ctx context.Context, name string) error {
+	return drain.RunNodeDrain(&drain.Helper{
+		Client:              k.kubernetescli,
+		Force:               true,
+		GracePeriodSeconds:  -1,
+		IgnoreAllDaemonSets: true,
+		Timeout:             2 * time.Minute,
+		DeleteEmptyDirData:  true,
+		DisableEviction:     true,
+		OnPodDeletedOrEvicted: func(pod *corev1.Pod, usingEviction bool) {
+			k.log.Printf("deleted pod %s/%s", pod.Namespace, pod.Name)
+		},
+		Out:    k.log.Writer(),
+		ErrOut: k.log.Writer(),
+	}, name)
+}
```

Comment:
We're not actually using `KubeDrain` in this PR at all.  Should be removed?

Reply:
Agreed, remnant from when kubedrain was part of the process. This will be handled in a future GA specifically made for kubedrain actions